app/assets/javascripts/iframe-saver.coffee

```diff
@@ -82,30 +75,39 @@ class IFrameSaver
   error: (msg) ->
     @save_indicator.showSaveFailed(msg)
 
-  save: (success_callback=null) ->
+  save: (success_callback = null) ->
     @success_callback = success_callback
 
     # will call back into "@save_to_server)
-    @iframePhone.post({ type:'getInteractiveState' })
+    @iframePhone.post({type: 'getInteractiveState'})
 
   confirm_delete: (callback) ->
     if (window.confirm("Are you sure you want to restart your work in this model?"))
       callback()
 
   delete_data: () ->
+    # Disable autosave, as it's possible that autosave will be triggered *after* we send to server "null" state
+    # (delete it). Actually it used to happen quite ofted.
```

Comment:
We don't have to re-enable autosave, as we reload the page at the end. However I'm using the same function to enable autosave in the constructor. Not sure if that's what you meant.

Re auto-save as an edge case - what about the last page of an activity, where there is no "next" button that forces Iframe saving? I asked about that through Github some time ago.
Single page has the same issues as the last page, so that's why I've added couple of additional autosave scenarios (focus, mouseout and changed 42 seconds to 5).